src/main/java/net/openhft/chronicle/queue/impl/single/SCQIndexing.java

```diff
@@ -194,18 +195,23 @@ private LongArrayValues arrayForAddress(@NotNull Wire wire, long secondaryAddres
 
     @NotNull
     private LongArrayValues array(@NotNull WireIn w, @NotNull LongArrayValues using, boolean index2index) {
-        final StringBuilder sb = Wires.acquireStringBuilder();
-        long readPos = w.bytes().readPosition();
-        @NotNull final ValueIn valueIn = w.readEventName(sb);
-        String name = index2index ? "index2index" : "index";
-        if (!name.contentEquals(sb))
-            throw new IllegalStateException("expecting index, was " + sb + ", bytes: " + w.bytes().readPosition(readPos).toHexString());
-
+        @NotNull final ValueIn valueIn = readIndexValue(w, index2index ? "index2index" : "index");
         valueIn.int64array(using, this, (o1, o2) -> {
         });
         return using;
     }
 
+    private ValueIn readIndexValue(@NotNull WireIn w, @NotNull String expectedName) {
+        try (ScopedResource<StringBuilder> stlSb = Wires.acquireStringBuilderScoped()) {
+            final StringBuilder sb = stlSb.get();
+            long readPos = w.bytes().readPosition();
+            @NotNull final ValueIn valueIn = w.readEventName(sb);
+            if (!expectedName.contentEquals(sb))
+                throw new IllegalStateException("expecting " + expectedName + ", was " + sb + ", bytes: " + w.bytes().readPosition(readPos).toHexString());
+            return valueIn;
+        }
+    }
```

Comment:
This pattern is repeated a few times, I wonder if `WireIn` should provide something like
```java
   public ValueIn expectEvent(String expectedName) throws UnexpectedValueException
```